### PR TITLE
Graham/add user

### PR DESCRIPTION
**Pull request by gbooth27**
Adds user tracking functionality to enable https://github.com/outline-insurance/outline2.0/pull/5907

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on: 
+  push:
+    branches:
+      - main
+
+jobs:
+  publish:
```
Comment (gbooth27) [inline .github/workflows/npm-publish.yml:7]: should just auto publish for us

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+on: 
+  push:
+    branches:
+      - main
```
Comment (iwong3) [inline .github/workflows/npm-publish.yml:4]: do we need to change this?

Comment (gbooth27) [reply .github/workflows/npm-publish.yml:4]: we could, was leaving as much the same as I could to facilitate somewhat easier merges in the future if we want to pull in changes from what we forked this from

Comment (iwong3) [reply .github/workflows/npm-publish.yml:4]: oh right this isn't our repo

**Review by iwong3 — APPROVED** [review]
LGTM

**Pull request merged**